### PR TITLE
remove babel from build and tests, target es2019 and add cjs

### LINKED ISSUE #68: New builds break create-react-app in 4.0.4-beta
Not sure when introduced, but the "nullish coalescing operator" (`??`) from TS and ES2020 breaks `create-react-app` builds.

The way to build for this with tsc is to lower your target-build to ES2019. With Babel, this is more complicated.

I don't think we should babel-transform these out of the library, I think we should remove babel from our build pipeline entirely.

I will make PR that relies only on `tsc`, and also adds a commonjs build for nodejs users, as detailed here:
https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html

### PR DESCRIPTION
**Pull request by dghelm**
Fixes create-react-app compilation and greatly simplifies our build pipeline. Removes all babel dependencies, moves to ts-jest for testing and updates package info.

Closes #68, #66, #9

### CODE REVIEW

**Comment by m-cat** [comment]
Cool. This also fixes https://github.com/SkynetLabs/skynet-js/issues/9 right?

**Comment by dghelm** [comment]
> Cool. This also fixes #9 right?

You're right! I think it had been fixed using another method, but so long as no one objects, getting rid of any babel seems really nice as a way to reduce how much documentation and tooling our lib touches.

File: .github/workflows/npmpublish.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v1
         with:
-          node-version: 12
+          node-version: 14
```
Comment (m-cat) [inline .github/workflows/npmpublish.yml:17]: Why not use 16 here? 🤔🤔

File: package.json [hunk 2]
```diff
@@ -2,7 +2,13 @@
   "name": "skynet-js",
   "version": "4.0.4-beta",
   "description": "Sia Skynet Javascript Client",
-  "main": "dist/index.js",
+  "type": "module",
+  "main": "dist/cjs/entry.js",
+  "module": "dist/mjs/index.js",
+  "exports": {
```
Comment (m-cat) [inline package.json:8]: Can you explain this please? Does this PR add `js/` and `mjs/` directories?

File: .github/workflows/npmpublish.yml (hunk 1 above)
Comment (dghelm) [reply .github/workflows/npmpublish.yml]: Fixed!

File: package.json (hunk 2 above)
Comment (dghelm) [reply package.json:8]: So "module" format, or ESM will be in `./dist/mjs/` and commonJS in `./dist/cjs/` -- the package.json should point projects in node or browserjs to the correct dir for their import types. I think this also means when we modify browser-specific code to node, we can just target the commonjs build.

I'm following the lead of https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html and a few other assorted resources.

Comment (dghelm) [reply package.json:8]: There is no longer a `./dist/js` dir, but you will see right away all the `require` statements in the cjs files.

Comment (dghelm) [reply package.json:7]: Misunderstood you initial comment, but this should be resolved by matching the article in commit 11537c0

**Review by m-cat — APPROVED** [review]

**Review by m-cat — CHANGES_REQUESTED** [review]
So I tried to build `mysky` with skynet-js linked and I still had your tsc-only branch built there. I kept seeign this error:
```
      TS2307: Cannot find module 'skynet-js' or its corresponding type declarations.
```
I switch to a different branch, built, and it worked again

**Comment by dghelm** [comment]
> So I tried to build `mysky` with skynet-js linked and I still had your tsc-only branch built there. I kept seeign this error:
> 
> ```
>       TS2307: Cannot find module 'skynet-js' or its corresponding type declarations.
> ```
> 
> I switch to a different branch, built, and it worked again

`"main": "dist/cjs/entry.js",` was supposed to be index.js. Also added "types" since ts docs say an explicit declaration is best practice.

Could run webpack build on `mysky` and successfully built TS Create React App and CRA projects.

**Review by kwypchlo — APPROVED** [review]
Looks good, make sure to confirm that typing works when using in a typescript project and that it works with webpack 4 and 5, create-raact-app starter, gatsby starter and nextjs starter.

**Review by m-cat — APPROVED** [review]
Works for me now!

**Pull request merged**